servers/rendering/renderer_rd/environment/sky.cpp

```diff
@@ -1020,11 +1013,19 @@ SkyRD::~SkyRD() {
 	RD::get_singleton()->free(index_buffer); //array gets freed as dependency
 }
 
-void SkyRD::setup(RID p_env, Ref<RenderSceneBuffersRD> p_render_buffers, const PagedArray<RID> &p_lights, RID p_camera_attributes, const Projection &p_projection, const Transform3D &p_transform, const Size2i p_screen_size, RendererSceneRenderRD *p_scene_render) {
+void SkyRD::setup_sky(RID p_env, Ref<RenderSceneBuffersRD> p_render_buffers, const PagedArray<RID> &p_lights, RID p_camera_attributes, uint32_t p_view_count, const Projection *p_view_projections, const Vector3 *p_view_eye_offsets, const Transform3D &p_cam_transform, const Size2i p_screen_size, RendererSceneRenderRD *p_scene_render) {
 	RendererRD::LightStorage *light_storage = RendererRD::LightStorage::get_singleton();
 	RendererRD::MaterialStorage *material_storage = RendererRD::MaterialStorage::get_singleton();
 	ERR_FAIL_COND(p_env.is_null());
 
+	// TODO FIXME, reflection probes currently don't have render buffers, but we need them for sky to work, implement for reflection probes!
+	// DO NOT MERGE UNTIL RESOLVED!!
+	ERR_FAIL_COND(p_render_buffers.is_null());
+
+	// make sure we support our view count
+	ERR_FAIL_COND(p_view_count == 0);
```

Comment:
I wonder if we should actually return a true/false from `setup_sky`, then we can do our checks in one place and prevent the rest from running on failure. Right now it is possibly we run other stages of sky rendering if setup fails.

Reply:
Thats a good idea, we could likely set ``draw_sky`` to false if ``setup_sky`` fails which would avoid the other operations. Thats something for another PR though as we would need to think about what kind of fallback information we would need to provide in that case (clear color etc.).